src/pages/home/report/ReportActionCompose/ComposerWithSuggestions.js

```diff
@@ -0,0 +1,569 @@
+        }
+
+        updateComment('');
+        setTextInputShouldClear(true);
+        if (isComposerFullSize) {
+            Report.setIsComposerFullSize(reportID, false);
+        }
+        setIsFullComposerAvailable(false);
+        return trimmedComment;
+    }, [updateComment, setTextInputShouldClear, isComposerFullSize, setIsFullComposerAvailable, reportID]);
+
+    /**
+     * Callback to add whatever text is chosen into the main input (used f.e as callback for the emoji picker)
+     * @param {String} text
+     * @param {Boolean} shouldAddTrailSpace
+     */
+    const replaceSelectionWithText = useCallback(
+        (text, shouldAddTrailSpace = true) => {
+            const updatedText = shouldAddTrailSpace ? `${text} ` : text;
+            const selectionSpaceLength = shouldAddTrailSpace ? CONST.SPACE_LENGTH : 0;
+            updateComment(ComposerUtils.insertText(commentRef.current, selection, updatedText));
+            setSelection((prevSelection) => ({
+                start: prevSelection.start + text.length + selectionSpaceLength,
+                end: prevSelection.start + text.length + selectionSpaceLength,
+            }));
+        },
+        [selection, updateComment],
+    );
+
+    const triggerHotkeyActions = useCallback(
+        (e) => {
+            if (!e || ComposerUtils.canSkipTriggerHotkeys(isSmallScreenWidth, isKeyboardShown)) {
+                return;
+            }
+
+            if (suggestionsRef.current.triggerHotkeyActions(e)) {
+                return;
+            }
+
+            // Submit the form when Enter is pressed
+            if (e.key === CONST.KEYBOARD_SHORTCUTS.ENTER.shortcutKey && !e.shiftKey) {
+                e.preventDefault();
+                submitForm();
+            }
+
+            // Trigger the edit box for last sent message if ArrowUp is pressed and the comment is empty and Chronos is not in the participants
+            const valueLength = valueRef.current.length;
+            if (e.key === CONST.KEYBOARD_SHORTCUTS.ARROW_UP.shortcutKey && textInputRef.current.selectionStart === 0 && valueLength === 0 && !ReportUtils.chatIncludesChronos(report)) {
+                e.preventDefault();
+
+                const parentReportActionID = lodashGet(report, 'parentReportActionID', '');
+                const parentReportAction = lodashGet(parentReportActions, [parentReportActionID], {});
+                const lastReportAction = _.find([...reportActions, parentReportAction], (action) => ReportUtils.canEditReportAction(action));
+
+                if (lastReportAction !== -1 && lastReportAction) {
+                    Report.saveReportActionDraft(reportID, lastReportAction.reportActionID, _.last(lastReportAction.message).html);
+                }
+            }
+        },
+        [isKeyboardShown, isSmallScreenWidth, parentReportActions, report, reportActions, reportID, submitForm, suggestionsRef, valueRef],
+    );
+
+    const onSelectionChange = useCallback(
```

Comment:
Coming from https://github.com/Expensify/App/issues/26534

We should've ensured to open the suggestion list only when the composer is focused, to prevent unexpected behavior.